.github/workflows/mlir-tensorrt-ci.yml

```diff
@@ -0,0 +1,227 @@
+                    stdout, _ = p.communicate()
+                    if p.returncode != 0:
+                        sys.exit(p.returncode)
+
+                    if not apply_diff:
+                        with open(filename) as f:
+                            code = f.readlines()
+                        formatted_code = StringIO(stdout).readlines()
+                        diff = difflib.unified_diff(
+                            code,
+                            formatted_code,
+                            filename,
+                            filename,
+                            "(before formatting)",
+                            "(after formatting)",
+                        )
+                        diff_string = "".join(diff)
+                        if len(diff_string) > 0:
+                            sys.stdout.write(diff_string)
+                            fail = True
+
+                if not apply_diff and fail:
+                    print("Style check failed, please apply the output diff.")
+                    sys.exit(1)
+
+
+            if __name__ == "__main__":
+                parser = argparse.ArgumentParser()
+                parser.add_argument(
+                    "--apply-diff",
+                    action="store_true",
+                    default=False,
+                    help="apply edits to files instead of displaying a diff",
+                )
+                args = parser.parse_args()
+                run_clang_format(args.apply_diff)
+            EOF
+
+            cat > run_format_check.sh <<EOF
+            #!/bin/bash
+            set -e
+            python3 -m black --check --exclude='.*\.pyi' mlir-tensorrt/test/
+            python3 -m black --check --exclude='.*\.pyi' mlir-tensorrt/python/
+            python3 run_clang_style_check.py
+            EOF
+
+            bash run_format_check.sh
+
+      # Create cache folders
+      - name: Create cache folder
+        run: |
+          mkdir -p ${{ github.workspace }}/ccache
+          mkdir -p ${{ github.workspace }}/.ccache.cpm
+
+      # Create cache action
+      - name: Create cache action
+        id: core-build-cache
+        uses: actions/cache@v4
+        with:
+          key: ${{ runner.os }}-mlir-tensorrt-core-build
+          path: |
+            ${{ github.workspace }}/ccache
+            ${{ github.workspace }}/.ccache.cpm
+
+      # Run LIT tests
+      - name: Run MLIR-TensorRT lit tests
+        continue-on-error: true
+        uses: addnab/docker-run-action@v3
+        with:
+          image: ${{ env.selected_test_image }}
+          options: -v ${{ github.workspace }}/mlir-tensorrt:/mlir-tensorrt -v ${{ github.workspace }}/ccache:/ccache -v ${{ github.workspace }}/.ccache.cpm:/.ccache.cpm
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+          run: |
+            export CCACHE_BASEDIR="$PWD"
+            export CCACHE_DIR="$PWD/ccache"
```

Comment:
Does this actually match `${{ github.workspace}}/ccache`?

Reply:
It does work since I mount these inside docker before.

https://github.com/NVIDIA/TensorRT-Incubator/actions/runs/10514361515/job/29132010233#step:12:3579

GitHub cache action takes care of lading and saving.
Without cache CI runs take ~1:15 minutes. Mainly because of LLVM build.
With cache CI run takes ~8 minutes e2e.